### PR TITLE
KIN-336 Add files managament in categories crud

### PR DESCRIPTION
**Pull request by Cristianpan**
[KIN-336](https://loopcrack.atlassian.net/browse/KIN-336) Medium [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=loopcrack-org&utm_campaign=badge_info)

### Problem

Administrators must be able to add and update category files.

### Solution

Implement the files management logic in the categories crud. 

### Requirements checklist

On create category:

- [x] It must store the category files on the server and files routes on the database. Finally link them in category record. 

On edit category:

- [x] If the edit category image and/or icon are changed, the new files must be stored on the server and their routes stored in the database, as well as link this new file records to the category record. Finally the old files must be deleted from the server and the database.
- [x] It is important to first register the new files and then delete the old ones because of the database design. 

On delete category:

- [x] It must delete the category files from the server and the database.

### Testing
_Note:_ Before to test this feature, run in console `php spark migrate:refresh` and as next step `php spark db:seed MainTestSeeder` to update the categories table. 

- [x] Create a new category and add its corresponding files. Verify that the files are saved on the server and in the db. 
- [x] Updates the files in an existing category and verifies that the new files are saved on the server and in the database, as well as deleting the old files from the server and db. 
- [x] Deletes a category and verifies that the files are removed from the server as well as from the db.

## Demo

[KIN-336]: https://loopcrack.atlassian.net/browse/KIN-336?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Oswaldo-Chan — APPROVED** [review]

**Review by CheloChidoone — APPROVED** [review]
Good Job @Cristianpan and thanks for adding the category subname field

**Review by CarlosJCD — APPROVED** [review]
Nice work!

File: app/Controllers/CtrlCategory.php [hunk 1]
```diff
@@ -2,93 +2,114 @@
...
 use App\Models\CategoryModel;
 use App\Models\CategoryTagModel;
+use App\Utils\FileManager;
 use App\Validation\CategoryValidation;
 use Throwable;
 
-class CtrlCategory extends BaseController
+class CtrlCategory extends CtrlApiFiles
 {
+    private const CATEGORIES_BASE_ROUTE = '/admin/categorias/';
```
Comment (FernandoJoachin) [inline app/Controllers/CtrlCategory.php:16]: You should follow the new standard we are using to create class attributes over route constants. Check CtrlSolution

File: app/Controllers/CtrlCategory.php [hunk 2]
```diff
@@ -2,93 +2,114 @@
...
+
+    protected FileValidationConfig $fileConfig;
+
+    public function __construct()
+    {
+        $fileConfigBuilder = new FileValidationConfigBuilder('/admin/categorias');
```
Comment (FernandoJoachin) [inline app/Controllers/CtrlCategory.php:22]: Don't forget to change this to class attributes

**Review by FernandoJoachin — CHANGES_REQUESTED** [review]
Great job!, just make those small changes.

**Review by FernandoJoachin — APPROVED** [review]

**Pull request merged**